aten/src/ATen/native/cuda/UpSampleNearest1d.cu

```diff
@@ -106,35 +96,33 @@ static void upsample_nearest1d_out_cuda_template(
 
   int output_width = output_size[0];
 
-  int nbatch = input.size(0);
-  int channels = input.size(1);
-  int input_width = input.size(2);
+  int nbatch = input_.size(0);
+  int channels = input_.size(1);
+  int input_width = input_.size(2);
 
   upsample_1d_shape_check(
-      input, Tensor(), nbatch, channels, input_width, output_width);
+      input_, Tensor(), nbatch, channels, input_width, output_width);
 
   AT_ASSERT(input_width > 0 && output_width > 0);
 
+  Tensor input = input_.contiguous();
   output.resize_({input.size(0), input.size(1), output_width});
   output.zero_();
 
-  const int num_kernels = output_width;
-  const int num_threads = std::min(
-      at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
+  unsigned int n = output.numel() / nbatch;
```

Comment:
Is there a guard anywhere against nbatch and output.numel being 0? If not, it will lead to fpe. If there is, may be add a comment so that it's clear.

Reply:
It's already done in the `upsample_xd_shape_check` earlier: https://github.com/pytorch/pytorch/blob/74bdcd44c4ea52896e56aef62369168abbf4a48e/aten/src/ATen/native/cuda/UpSample.cuh#L21-L124
I'll put a comment.